### PR TITLE
harlequin: init at 1.24.1

### PR DESCRIPTION
**Pull request by pcboy**
## Description of changes

Following package request:
https://github.com/NixOS/nixpkgs/issues/280605

Harlequin is a SQL IDE in your terminal. It's a python CLI app.

## Things done

Harlequin supports a lot of different adapters today.
This PR only supports duckdb, sqlite, postgres, and bigquery. More can be added but this is quite some work. So would prefer to first make sure if it's necessary or not for first merging.

This is also the first time I had to make a python package with optional dependencies (`passthru.optional-dependencies`). I'm not sure if what I did is correct and I could not find much guidelines about it. :thinking: 

I also had to remove the `harlequin` dep from `harlequin-postgres` and `harlequin-bigquery` to prevent circular dependencies.

I tested it on my machine (NixOS x86_64-linux) and harlequin seems to work successfully.

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/by-name/ha/harlequin/package.nix [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    #   harlequin-odbc
+    # ];
+    # s3 = [
+    #   boto3
+    # ];
+    # trino = [
+    #   harlequin-trino
+    # ];
+  };
+
```
Comment (GaetanLepage) [inline pkgs/by-name/ha/harlequin/package.nix:76]:
```suggestion
```

You can just remove those, they are not very useful.
Especially regarding the fact that this is packaged as a top-level application and not a package.

File: pkgs/by-name/ha/harlequin/package.nix [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  ];
+
+  pythonImportsCheck = [
+    "harlequin"
+    "harlequin_duckdb"
+    "harlequin_sqlite"
+    "harlequin_vscode"
+  ];
+
+  meta = {
```
Comment (GaetanLepage) [inline pkgs/by-name/ha/harlequin/package.nix:88]: You should add `mainProgram` to indicate the name of the main executable provided by this package.

File: pkgs/development/python-modules/sqlfmt/default.nix [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+buildPythonPackage rec {
+  pname = "sqlfmt";
+  version = "0.23.2";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "tconbeer";
+    repo = "sqlfmt";
+    rev = "v${version}";
```
Comment (GaetanLepage) [inline pkgs/development/python-modules/sqlfmt/default.nix:25]:
```suggestion
    rev = "refs/tags/v${version}";
```
This prevents branch/tag mismatch.

File: pkgs/by-name/ha/harlequin/package.nix [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+}:
+python3.pkgs.buildPythonApplication rec {
+  pname = "harlequin";
+  version = "1.24.0";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "tconbeer";
+    repo = "harlequin";
+    rev = "v${version}";
```
Comment (GaetanLepage) [inline pkgs/by-name/ha/harlequin/package.nix:16]:
```suggestion
    rev = "refs/tags/v${version}";
```

**Review by GaetanLepage — CHANGES_REQUESTED** [review]
Minor comments. It looks good overall.

**Comment by pcboy** [comment]
@GaetanLepage Thanks for the review!

I updated with all your suggestions. :+1:

File: pkgs/by-name/ha/harlequin/package.nix [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+  ];
+
+  pythonImportsCheck = [
+    "harlequin"
+    "harlequin_duckdb"
+    "harlequin_sqlite"
+    "harlequin_vscode"
+  ];
+
+  meta = {
```
Comment (GaetanLepage) [inline pkgs/by-name/ha/harlequin/package.nix:55]:
```suggestion
  ];
  
  passthru = {
    updateScript = nix-update-script { };
    tests.version = testers.testVersion { package = harlequin; };
  };

  meta = {
```
Don't forget to add `harlequin` to the derivation inputs.

Comment (pcboy) [reply pkgs/by-name/ha/harlequin/package.nix]: Thank you. I added it, but surprisingly, I'm encountering issues.

```
took 4s ➜  nix-build -A harlequin.tests.version                     
this derivation will be built:
  /nix/store/slcc8v51d4cwli39sssv1riryqhfxl3m-harlequin-1.24.0-test-version.drv
building '/nix/store/slcc8v51d4cwli39sssv1riryqhfxl3m-harlequin-1.24.0-test-version.drv'...
harlequin --version returned a non-zero exit code.
Harlequin could not load the installed plug-in named google.cloud.

No module named 'google.cloud'
Traceback (most recent call last):
  File "{{storeDir}}/bin/.harlequin-wrapped", line 9, in <module>
    sys.exit(harlequin())
             ^^^^^^^^^^^
  File "{{storeDir}}/lib/python3.12/site-packages/harlequin/cli.py", line 413, in harlequin
    cli = build_cli()
          ^^^^^^^^^^^
  File "{{storeDir}}/lib/python3.12/site-packages/harlequin/cli.py", line 153, in build_cli
    adapters = load_adapter_plugins()
               ^^^^^^^^^^^^^^^^^^^^^^
  File "{{storeDir}}/lib/python3.12/site-packages/harlequin/plugins.py", line 17, in load_adapter_plugins
    return _load_plugins(group="harlequin.adapter")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "{{storeDir}}/lib/python3.12/site-packages/harlequin/plugins.py", line 62, in _load_plugins
    return plugins
           ^^^^^^^
UnboundLocalError: cannot access local variable 'plugins' where it is not associated with a value
```

This is strange as if I just `nix-build -A harlequin` and try to use it with `harlequin -a bigquery` it works perfectly fine. :thinking: 

What am I missing?

Comment (GaetanLepage) [reply pkgs/by-name/ha/harlequin/package.nix]: This test is simply running `harlequin --version`.
Is this command supposed to exist ? Currently, it is failing with the message you are seeing.

Comment (pcboy) [reply pkgs/by-name/ha/harlequin/package.nix]: Yes the command works fine if I do `nix run .\#harlequin -- --version`

```
harlequin, version 1.24.0

Installed Adapters:
  - duckdb, version 1.24.0
  - sqlite, version 1.24.0
  - postgres, version 0.4.0
  - bigquery, version 1.0.2
```

File: pkgs/by-name/ha/harlequin/package.nix [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+  fetchFromGitHub,
+  harlequin,
+  testers,
+  nix-update-script,
+  withPostgresAdapter ? true,
+  withBigQueryAdapter ? true,
+}:
+python3.pkgs.buildPythonApplication rec {
+  pname = "harlequin";
+  version = "1.24.0";
```
Comment (GaetanLepage) [inline pkgs/by-name/ha/harlequin/package.nix:13]:
```suggestion
  version = "1.24.1";
```
1.24.1 is out.

File: pkgs/by-name/ha/harlequin/package.nix [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "tconbeer";
+    repo = "harlequin";
+    rev = "ref/tags/v${version}";
```
Comment (GaetanLepage) [inline pkgs/by-name/ha/harlequin/package.nix:19]:
```suggestion
    rev = "refs/tags/v${version}";
```

**Review by GaetanLepage — CHANGES_REQUESTED** [review]

File: pkgs/by-name/ha/harlequin/package.nix [hunk 8]
```diff
@@ -0,0 +1,75 @@
+{
+  lib,
+  python3,
```
Comment (GaetanLepage) [inline pkgs/by-name/ha/harlequin/package.nix:3]:
```suggestion
  python3Packages,
```

File: pkgs/by-name/ha/harlequin/package.nix [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+  fetchFromGitHub,
+  harlequin,
+  testers,
+  nix-update-script,
+  withPostgresAdapter ? true,
+  withBigQueryAdapter ? true,
+}:
+python3.pkgs.buildPythonApplication rec {
```
Comment (GaetanLepage) [inline pkgs/by-name/ha/harlequin/package.nix:11]:
```suggestion
python3Packages.buildPythonApplication rec {
```

File: pkgs/by-name/ha/harlequin/package.nix [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "tconbeer";
+    repo = "harlequin";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-3Rb47zkWsC6RJhk1btQc/kwxpFFWVnxY2PJooHB7IzQ=";
+  };
+
+  build-system = with python3.pkgs; [
```
Comment (GaetanLepage) [inline pkgs/by-name/ha/harlequin/package.nix:23]:
```suggestion
  build-system = with python3Packages; [
```

File: pkgs/by-name/ha/harlequin/package.nix [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+    poetry-core
+  ];
+
+  dependencies =
+    with python3.pkgs;
```
Comment (GaetanLepage) [inline pkgs/by-name/ha/harlequin/package.nix:28]:
```suggestion
    with python3Packages;
```

File: pkgs/by-name/ha/harlequin/package.nix [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+
+  passthru = {
+    updateScript = nix-update-script { };
+    tests.version = testers.testVersion {
+      package = harlequin;
+      command = "harlequin --version";
+    };
+  };
+
+  meta = {
```
Comment (GaetanLepage) [inline pkgs/by-name/ha/harlequin/package.nix:66]:
```suggestion
  };
  
  nativeCheckInputs = [
    versionCheckHook
  ];

  meta = {
```

**Review by GaetanLepage — CHANGES_REQUESTED** [review]

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 343967`

---
### `x86_64-linux`
<details>
  <summary>:x: 7 packages failed to build:</summary>
  <ul>
    <li>python311Packages.harlequin-bigquery</li>
    <li>python311Packages.harlequin-postgres</li>
    <li>python311Packages.harlequin-postgres.dist</li>
    <li>python311Packages.textual-fastdatatable</li>
    <li>python311Packages.textual-fastdatatable.dist</li>
    <li>python311Packages.textual-textarea</li>
    <li>python311Packages.textual-textarea.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 15 packages built:</summary>
  <ul>
    <li>harlequin</li>
    <li>harlequin.dist</li>
    <li>python311Packages.harlequin-bigquery.dist</li>
    <li>python311Packages.sqlfmt</li>
    <li>python311Packages.sqlfmt.dist</li>
    <li>python312Packages.harlequin-bigquery</li>
    <li>python312Packages.harlequin-bigquery.dist</li>
    <li>python312Packages.harlequin-postgres</li>
    <li>python312Packages.harlequin-postgres.dist</li>
    <li>python312Packages.sqlfmt</li>
    <li>python312Packages.sqlfmt.dist</li>
    <li>python312Packages.textual-fastdatatable</li>
    <li>python312Packages.textual-fastdatatable.dist</li>
    <li>python312Packages.textual-textarea</li>
    <li>python312Packages.textual-textarea.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:x: 14 packages failed to build:</summary>
  <ul>
    <li>harlequin</li>
    <li>harlequin.dist</li>
    <li>python311Packages.harlequin-bigquery</li>
    <li>python311Packages.harlequin-bigquery.dist</li>
    <li>python311Packages.harlequin-postgres</li>
    <li>python311Packages.textual-fastdatatable</li>
    <li>python311Packages.textual-textarea</li>
    <li>python311Packages.textual-textarea.dist</li>
    <li>python312Packages.harlequin-bigquery</li>
    <li>python312Packages.harlequin-bigquery.dist</li>
    <li>python312Packages.harlequin-postgres</li>
    <li>python312Packages.harlequin-postgres.dist</li>
    <li>python312Packages.textual-textarea</li>
    <li>python312Packages.textual-textarea.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 8 packages built:</summary>
  <ul>
    <li>python311Packages.harlequin-postgres.dist</li>
    <li>python311Packages.sqlfmt</li>
    <li>python311Packages.sqlfmt.dist</li>
    <li>python311Packages.textual-fastdatatable.dist</li>
    <li>python312Packages.sqlfmt</li>
    <li>python312Packages.sqlfmt.dist</li>
    <li>python312Packages.textual-fastdatatable</li>
    <li>python312Packages.textual-fastdatatable.dist</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:x: 21 packages failed to build:</summary>
  <ul>
    <li>harlequin</li>
    <li>harlequin.dist</li>
    <li>python311Packages.harlequin-bigquery</li>
    <li>python311Packages.harlequin-bigquery.dist</li>
    <li>python311Packages.harlequin-postgres</li>
    <li>python311Packages.sqlfmt</li>
    <li>python311Packages.sqlfmt.dist</li>
    <li>python311Packages.textual-fastdatatable</li>
    <li>python311Packages.textual-fastdatatable.dist</li>
    <li>python311Packages.textual-textarea</li>
    <li>python311Packages.textual-textarea.dist</li>
    <li>python312Packages.harlequin-bigquery</li>
    <li>python312Packages.harlequin-bigquery.dist</li>
    <li>python312Packages.harlequin-postgres</li>
    <li>python312Packages.harlequin-postgres.dist</li>
    <li>python312Packages.sqlfmt</li>
    <li>python312Packages.sqlfmt.dist</li>
    <li>python312Packages.textual-fastdatatable</li>
    <li>python312Packages.textual-fastdatatable.dist</li>
    <li>python312Packages.textual-textarea</li>
    <li>python312Packages.textual-textarea.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>python311Packages.harlequin-postgres.dist</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:x: 21 packages failed to build:</summary>
  <ul>
    <li>harlequin</li>
    <li>harlequin.dist</li>
    <li>python311Packages.harlequin-bigquery</li>
    <li>python311Packages.harlequin-bigquery.dist</li>
    <li>python311Packages.harlequin-postgres</li>
    <li>python311Packages.sqlfmt</li>
    <li>python311Packages.sqlfmt.dist</li>
    <li>python311Packages.textual-fastdatatable</li>
    <li>python311Packages.textual-fastdatatable.dist</li>
    <li>python311Packages.textual-textarea</li>
    <li>python311Packages.textual-textarea.dist</li>
    <li>python312Packages.harlequin-bigquery</li>
    <li>python312Packages.harlequin-bigquery.dist</li>
    <li>python312Packages.harlequin-postgres</li>
    <li>python312Packages.harlequin-postgres.dist</li>
    <li>python312Packages.sqlfmt</li>
    <li>python312Packages.sqlfmt.dist</li>
    <li>python312Packages.textual-fastdatatable</li>
    <li>python312Packages.textual-fastdatatable.dist</li>
    <li>python312Packages.textual-textarea</li>
    <li>python312Packages.textual-textarea.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>python311Packages.harlequin-postgres.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 343967`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 22 packages built:</summary>
  <ul>
    <li>harlequin</li>
    <li>harlequin.dist</li>
    <li>python311Packages.harlequin-bigquery</li>
    <li>python311Packages.harlequin-bigquery.dist</li>
    <li>python311Packages.harlequin-postgres</li>
    <li>python311Packages.harlequin-postgres.dist</li>
    <li>python311Packages.sqlfmt</li>
    <li>python311Packages.sqlfmt.dist</li>
    <li>python311Packages.textual-fastdatatable</li>
    <li>python311Packages.textual-fastdatatable.dist</li>
    <li>python311Packages.textual-textarea</li>
    <li>python311Packages.textual-textarea.dist</li>
    <li>python312Packages.harlequin-bigquery</li>
    <li>python312Packages.harlequin-bigquery.dist</li>
    <li>python312Packages.harlequin-postgres</li>
    <li>python312Packages.harlequin-postgres.dist</li>
    <li>python312Packages.sqlfmt</li>
    <li>python312Packages.sqlfmt.dist</li>
    <li>python312Packages.textual-fastdatatable</li>
    <li>python312Packages.textual-fastdatatable.dist</li>
    <li>python312Packages.textual-textarea</li>
    <li>python312Packages.textual-textarea.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 22 packages built:</summary>
  <ul>
    <li>harlequin</li>
    <li>harlequin.dist</li>
    <li>python311Packages.harlequin-bigquery</li>
    <li>python311Packages.harlequin-bigquery.dist</li>
    <li>python311Packages.harlequin-postgres</li>
    <li>python311Packages.harlequin-postgres.dist</li>
    <li>python311Packages.sqlfmt</li>
    <li>python311Packages.sqlfmt.dist</li>
    <li>python311Packages.textual-fastdatatable</li>
    <li>python311Packages.textual-fastdatatable.dist</li>
    <li>python311Packages.textual-textarea</li>
    <li>python311Packages.textual-textarea.dist</li>
    <li>python312Packages.harlequin-bigquery</li>
    <li>python312Packages.harlequin-bigquery.dist</li>
    <li>python312Packages.harlequin-postgres</li>
    <li>python312Packages.harlequin-postgres.dist</li>
    <li>python312Packages.sqlfmt</li>
    <li>python312Packages.sqlfmt.dist</li>
    <li>python312Packages.textual-fastdatatable</li>
    <li>python312Packages.textual-fastdatatable.dist</li>
    <li>python312Packages.textual-textarea</li>
    <li>python312Packages.textual-textarea.dist</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 22 packages built:</summary>
  <ul>
    <li>harlequin</li>
    <li>harlequin.dist</li>
    <li>python311Packages.harlequin-bigquery</li>
    <li>python311Packages.harlequin-bigquery.dist</li>
    <li>python311Packages.harlequin-postgres</li>
    <li>python311Packages.harlequin-postgres.dist</li>
    <li>python311Packages.sqlfmt</li>
    <li>python311Packages.sqlfmt.dist</li>
    <li>python311Packages.textual-fastdatatable</li>
    <li>python311Packages.textual-fastdatatable.dist</li>
    <li>python311Packages.textual-textarea</li>
    <li>python311Packages.textual-textarea.dist</li>
    <li>python312Packages.harlequin-bigquery</li>
    <li>python312Packages.harlequin-bigquery.dist</li>
    <li>python312Packages.harlequin-postgres</li>
    <li>python312Packages.harlequin-postgres.dist</li>
    <li>python312Packages.sqlfmt</li>
    <li>python312Packages.sqlfmt.dist</li>
    <li>python312Packages.textual-fastdatatable</li>
    <li>python312Packages.textual-fastdatatable.dist</li>
    <li>python312Packages.textual-textarea</li>
    <li>python312Packages.textual-textarea.dist</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 22 packages built:</summary>
  <ul>
    <li>harlequin</li>
    <li>harlequin.dist</li>
    <li>python311Packages.harlequin-bigquery</li>
    <li>python311Packages.harlequin-bigquery.dist</li>
    <li>python311Packages.harlequin-postgres</li>
    <li>python311Packages.harlequin-postgres.dist</li>
    <li>python311Packages.sqlfmt</li>
    <li>python311Packages.sqlfmt.dist</li>
    <li>python311Packages.textual-fastdatatable</li>
    <li>python311Packages.textual-fastdatatable.dist</li>
    <li>python311Packages.textual-textarea</li>
    <li>python311Packages.textual-textarea.dist</li>
    <li>python312Packages.harlequin-bigquery</li>
    <li>python312Packages.harlequin-bigquery.dist</li>
    <li>python312Packages.harlequin-postgres</li>
    <li>python312Packages.harlequin-postgres.dist</li>
    <li>python312Packages.sqlfmt</li>
    <li>python312Packages.sqlfmt.dist</li>
    <li>python312Packages.textual-fastdatatable</li>
    <li>python312Packages.textual-fastdatatable.dist</li>
    <li>python312Packages.textual-textarea</li>
    <li>python312Packages.textual-textarea.dist</li>
  </ul>
</details>

**Review by GaetanLepage — APPROVED** [review]
LGTM !
Thanks for dealing with all my requests :)

**Comment by shinzui** [comment]
I was not able to install harlequin on `aarch64-darwin` 

```
error: builder for '/nix/store/swlpq3bihpgs0v1q68b89khp5cakd2kf-harlequin-1.25.2.drv' failed with exit code 1;
       last 25 log lines:
       > Running phase: unpackPhase
       > unpacking source archive /nix/store/59fzfh0k0vqg9bqz2xpywcqdcn19k4k0-source
       > source root is source
       > setting SOURCE_DATE_EPOCH to timestamp 315619200 of file source/tests/unit_tests/test_s3_tree.py
       > Running phase: patchPhase
       > Running phase: updateAutotoolsGnuConfigScriptsPhase
       > Running phase: configurePhase
       > no configure script, doing nothing
       > Running phase: buildPhase
       > Executing pypaBuildPhase
       > Creating a wheel...
       > pypa build flags: --no-isolation --outdir dist/ --wheel
       > * Getting build dependencies for wheel...
       > * Building wheel...
       > Successfully built harlequin-1.25.2-py3-none-any.whl
       > Finished creating a wheel...
       > /private/tmp/nix-build-harlequin-1.25.2.drv-0/source/dist /private/tmp/nix-build-harlequin-1.25.2.drv-0/source
       > Unpacking to: unpacked/harlequin-1.25.2...OK
       > Repacking wheel as ./harlequin-1.25.2-py3-none-any.whl...OK
       > /private/tmp/nix-build-harlequin-1.25.2.drv-0/source
       > Finished executing pypaBuildPhase
       > Running phase: pythonRuntimeDepsCheckHook
       > Executing pythonRuntimeDepsCheck
       > Checking runtime dependencies for harlequin-1.25.2-py3-none-any.whl
       >   - numpy<2.0.0,>=1.26.0 not satisfied by version 2.2.0
       For full logs, run 'nix log /nix/store/swlpq3bihpgs0v1q68b89khp5cakd2kf-harlequin-1.25.2.drv'
  ```

**Comment by uosis** [comment]
> I was not able to install harlequin on `aarch64-darwin`

Created build failure report: https://github.com/NixOS/nixpkgs/issues/370797

**Pull request merged**